src/Lean/Compiler/IR/EmitLLVM.lean

```diff
@@ -1529,6 +1529,26 @@ def optimizeLLVMModule (mod : LLVM.Module ctx) : IO Unit := do
   LLVM.disposePassManager pm
   LLVM.disposePassManagerBuilder pmb
 
+/-- Get the names of all global symbols in the module -/
+partial def getModuleGlobals (mod : LLVM.Module llvmctx) : IO (HashSet String) := do
+  let rec go (v : LLVM.Value llvmctx) (acc : HashSet String) : IO (HashSet String) := do
+    if v.isNull then return acc
+    else go (← LLVM.getNextGlobal v) (acc.insert (← LLVM.Value.getName v))
+  go (← LLVM.getFirstGlobal mod) {}
+
+/-- Get the names of all global functions in the module -/
+partial def getModuleFunctions (mod : LLVM.Module llvmctx) : IO (HashSet String) := do
```

Comment:
Nit before someone makes an update-stage0 for this: why a HashSet anyway?

Reply:
Changed to Array then re-update stage0